### PR TITLE
Team commendations

### LINKED ISSUE #84: Team Commendation Page
This is the implementation of the team commendation page. Keep in mind we already have the "commendation form" component, all this will take is a bit of doing to make it for teams rather than for individuals.

### PR DESCRIPTION
**Pull request by awtpi314**
Closes #84 

Adds team commendation page implementation.

### CODE REVIEW

File: lib/api/commendations/email.ts [hunk 1]
```diff
@@ -3,25 +3,29 @@ import { SendEmailCommandInput } from "@aws-sdk/client-ses/dist-types/commands";
 
 const client = new SES({ region: "us-east-2" });
 
-export const send_bz_email = async (senderEmail: string, recipientEmail: string, senderName: string, message: string) => {
+export const send_bz_email = async (
```
Comment (SpBills) [inline lib/api/commendations/email.ts:6]: change to `sendBzEmail`

File: pages/api/commendation/index.ts [hunk 2]
```diff
@@ -17,27 +18,57 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+        send_bz_email(
+          session?.user?.email as string,
+          recipients,
+          session?.user?.name as string,
+          msg
+        );
+      }
+
+      recipients.forEach(async (recipient) => {
+        const commendation = await createCommendation(sender as string, await emailToId(recipient) ?? "", msg);
```
Comment (SpBills) [inline pages/api/commendation/index.ts:68]: do we really get a list of emails from the frontend? we should not be getting a list of emails from the frontend.

Comment (SpBills) [reply pages/api/commendation/index.ts:68]: i guess it doesn't really matter?

File: components/TeamCommendationForm.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  };
+
+  return (
+    <Paper sx={{ mt: 4, mx: "auto", maxWidth: "30rem", p: 2 }}>
+      <form onSubmit={() => setSending(true)} action="api/commendation" method="POST">
+        <Stack spacing={1}>
+          <Typography color="primary" className={raleway.className} fontSize={25} fontWeight={900}>Create Commendation</Typography>
+          <TextField sx={{ display: "none" }} hidden name="recipient"
+            value={getTeam(teamData)?.members.map((currentMember) => {
+              return currentMember.email;
```
Comment (SpBills) [inline components/TeamCommendationForm.tsx:32]: send ids instead.

File: lib/api/commendations/crud.ts [hunk 4]
```diff
@@ -1,15 +1,11 @@
...
-  const student = await prisma.member.findFirst({ where: { id: studentId } });
-
-  if (!student) {
-    return "";
-  }
-
-  const { email } = student;
+export const idsToEmails = async (studentIds: string[]) => {
+  const emails = (await Promise.all(
+    studentIds.map((currentId) => prisma.member.findFirst({ where: { id: currentId } }))))
```
Comment (SpBills) [inline lib/api/commendations/crud.ts:5]: Do this as a bulk query instead of mapping in memory?

**Review by SpBills — APPROVED** [review]
Some general info-

Great work overall Alex! You did a really good job at making the business requirements work. There are some general architectural things.

1. You should *never* trust the client. Someone could send a bunch of emails, and it would send them without checking. (maybe? maybe not? either way, never trust the client to provide its own emails. we prefer sending the ids instead.)
2. Great job in general with the creating new functions.
3. We are trying to stay consistent with `camelCase` for functions and variables, and  `PascalCase` for components.
4. I wonder if it was necessary to make an entirely new component for the team commendation, and if we could have simply reused the single commendation. Nothing you have to change, just something to think about.

Overall, very good job. I am only criticizing because your work is very well done and I think you do a very good job at taking criticism and applying it. I hope this helps in the same way that you teaching me things about Prisma helped me. in past PRs.

**Review by SpBills — CHANGES_REQUESTED** [review]
oops, didn't mean to approve quite yet.

File: lib/api/teams/crud.ts [hunk 5]
```diff
@@ -1,9 +1,60 @@
...
+                    sentCommendations: { select: { id: true } }
+                }
+            }
+        },
+        orderBy: { name: "asc" }
+    })
+}
+
+export const getContactInfo = async (id: string) => {
+    const member = await prisma.member.findFirst({
```
Comment (SpBills) [inline lib/api/teams/crud.ts:19]: better hope that a member ID and a team ID never overlap.

File: lib/api/teams/crud.ts [hunk 6]
```diff
@@ -1,9 +1,60 @@
...
+        },
+        include: {
+            members: true
+        }
+    })
+
+    if (team) {
+        const emails: string[] = [];
+        const phoneNumbers: string[] = [];
+        team.members.forEach((currentMember) => {
```
Comment (SpBills) [inline lib/api/teams/crud.ts:41]: You should use map syntax instead here.

map takes `[T]` and turns it into type `[U]`, or in this case `[Member]` into `[string]`

yes, this means you'll have to do 2 separate maps.

**Review by SpBills — CHANGES_REQUESTED** [review]
mmm. much better. just a tiny change

File: lib/api/teams/crud.ts (hunk 5 above)
Comment (awtpi314) [reply lib/api/teams/crud.ts:19]: uuid cool

**Pull request merged**